tree/ntuple/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -392,18 +528,16 @@ private:
    /// \brief Construct a new RNTupleProcessor for processing a single RNTuple.
    ///
    /// \param[in] ntuple The source specification (name and storage location) for the RNTuple to process.
-   /// \param[in] model The model that specifies which fields should be read by the processor.
    /// \param[in] processorName Name of the processor. Unless specified otherwise in RNTupleProcessor::Create, this is
    /// the name of the underlying RNTuple.
-   RNTupleSingleProcessor(RNTupleOpenSpec ntuple, std::unique_ptr<ROOT::RNTupleModel> model,
-                          std::string_view processorName);
+   RNTupleSingleProcessor(RNTupleOpenSpec ntuple, std::string_view processorName);
 
 public:
    RNTupleSingleProcessor(const RNTupleSingleProcessor &) = delete;
    RNTupleSingleProcessor(RNTupleSingleProcessor &&) = delete;
    RNTupleSingleProcessor &operator=(const RNTupleSingleProcessor &) = delete;
    RNTupleSingleProcessor &operator=(RNTupleSingleProcessor &&) = delete;
-   ~RNTupleSingleProcessor() override { fModel.release(); };
+   ~RNTupleSingleProcessor() override { fProtoModel.release(); };
```

Comment:
Hm, why is `RNTupleSingleProcessor` releasing `RNTupleProcess::fProtoModel`? Does it have to do with the page source and destruction order? Is this still needed with the transition to a proto model?

Reply:
'Why' is it needed?

Reply:
Yeah I realized just now why: It's because otherwise the page source managed by the `RSingleProcessor` gets destructed first, and the fields in `fProtoModel` are connected to this page source, so it has to be destructed before.